src/AzureClient/AzureClient.cs

```diff
@@ -176,57 +187,89 @@ public async Task<ExecutionResult> GetConnectionStatusAsync(IChannel channel)
-
-                // TODO: Add encoder to visualize IEnumerable<KeyValuePair<string, double>>
-                return output.Histogram.ToExecutionResult();
+                entryPoint = EntryPointGenerator.Generate(operationName, ActiveTarget.TargetName);
             }
-            else
+            catch (UnsupportedOperationException e)
             {
-                channel.Stdout($"Submitting {operationName} to target {ActiveTarget.TargetName}...");
-                var job = await machine.SubmitAsync(entryPointInfo, entryPointInput);
-                channel.Stdout($"Job {job.Id} submitted successfully.");
+                channel.Stderr($"{operationName} is not a recognized Q# operation name.");
+                return AzureClientError.UnrecognizedOperationName.ToExecutionResult();
+            }
+            catch (CompilationErrorsException e)
+            {
+                channel.Stderr($"The Q# operation {operationName} could not be compiled as an entry point for job execution.");
+                foreach (var message in e.Errors) channel.Stderr(message);
+                return AzureClientError.InvalidEntryPoint.ToExecutionResult();
+            }
 
+            try
+            {
+                var job = await entryPoint.SubmitAsync(machine, inputParameters);
+                channel.Stdout($"Job {job.Id} submitted successfully.");
                 MostRecentJobId = job.Id;
+            }
+            catch (ArgumentException e)
+            {
+                channel.Stderr($"Failed to parse all expected parameters for Q# operation {operationName}.");
+                channel.Stderr(e.Message);
+                return AzureClientError.JobSubmissionFailed.ToExecutionResult();
+            }
+            catch (Exception e)
+            {
+                channel.Stderr($"Failed to submit Q# operation {operationName} for execution.");
+                channel.Stderr(e.InnerException?.Message ?? e.Message);
+                return AzureClientError.JobSubmissionFailed.ToExecutionResult();
+            }
 
-                // TODO: Add encoder for IQuantumMachineJob rather than calling ToJupyterTable() here.
-                return job.ToJupyterTable().ToExecutionResult();
+            if (!execute)
+            {
+                return await GetJobStatusAsync(channel, MostRecentJobId);
             }
+
+            var timeoutInSeconds = 30;
+            channel.Stdout($"Waiting up to {timeoutInSeconds} seconds for Azure Quantum job to complete...");
+
+            using (var cts = new System.Threading.CancellationTokenSource(TimeSpan.FromSeconds(30)))
+            {
+                CloudJob? cloudJob = null;
+                do
+                {
+                    // TODO: Once jupyter-core supports interrupt requests (https://github.com/microsoft/jupyter-core/issues/55),
+                    //       handle Jupyter kernel interrupt here and break out of this loop 
+                    var pollingIntervalInSeconds = 5;
```

Comment:
Similarly here.

Reply:
Same, this is configurable in https://github.com/microsoft/iqsharp/pull/161. Reading my mind...